pkg/cmd/stack/deploy.go

```diff
@@ -441,7 +441,11 @@ func getFailedStatusError(ctx context.Context, stack *model.Stack, svcName strin
 		for _, cStatus := range p.Status.ContainerStatuses {
 			totalRestarts += int(cStatus.RestartCount)
 		}
-		if totalRestarts >= maxRestartsToConsiderFailed {
+		maxSvcRestarts := stack.Services[dependingSvc].BackOffLimit
+		if maxSvcRestarts == 0 {
+			maxSvcRestarts = maxRestartsToConsiderFailed
```

Comment:
`maxRestartsToConsiderFailed` is the default max restarts to fail?

Reply:
yes, It's set to 3 by default on kubernetes resources